api/src/main/java/org/jboss/shrinkwrap/api/ArchiveFactory.java

```diff
@@ -183,20 +183,15 @@
                 + archiveFile.getAbsolutePath());
         }
 
-        // Construct ZipFile
-        final ZipFile zipFile;
         try {
-            zipFile = new ZipFile(archiveFile);
+            // Import
+            return ShrinkWrap.create(type, archiveFile.getName()).as(ZipImporter.class).importFrom(new ZipFile(archiveFile)).as(type);
```

Comment:
This was just inlining the new ZipFile as the variable is unused, and cannot be closed at this point..